feat: map layers in geolocation

0MUHAMMEDIRFAN opened the pull request:
This pull request enhances the geolocation and map view functionality by introducing multiple map layers (e.g., satellite, terrain, labels) and adding controls for switching between them. It also refactors the map defaults configuration to support these new layers. Below are the key changes grouped by theme:

### Geolocation Enhancements
* Added support for multiple map layers (`streetLayer`, `satelliteLayer`, `labelsLayer`, `terrainLayer`) in the `ControlGeolocation` class, with the default street layer being added to the map by default.
* Introduced a `bind_leaflet_layers_control` method to allow users to switch between base layers (e.g., Default, Satellite) and toggle overlays (e.g., Labels, Terrain). [[1]](diffhunk://#diff-da07e9816f1943d98641a8268c01df97534c2dbec4f115393ab949f4e686b82cR50) [[2]](diffhunk://#diff-da07e9816f1943d98641a8268c01df97534c2dbec4f115393ab949f4e686b82cL159-R200)
* Implemented logic to dynamically show or hide overlay options and remove overlays when switching between map views using the `baselayerchange` event.
* Added a utility method `display_leaflet_overlays_control` to control the visibility of overlay controls in the map interface.

### Map View Enhancements
* Extended the `MapView` class to support the same multi-layer functionality as geolocation, including adding base layers, overlays, and event listeners for managing layer visibility. [[1]](diffhunk://#diff-fa5ab989fe7a60abed8dc68ad7261f5aee51395ebacc6e685219b2306b32c534L46-R70) [[2]](diffhunk://#diff-fa5ab989fe7a60abed8dc68ad7261f5aee51395ebacc6e685219b2306b32c534R162-R222)

### Refactoring and Configuration Updates
* Refactored `frappe.utils.map_defaults.tiles` to support multiple tile configurations (e.g., default, satellite, labels, terrain) with corresponding URLs and options. This enables the dynamic addition of layers in both geolocation and map views.

These changes improve the user experience by providing richer map customization and interactivity features.Added labels option and tarrain lines option in sattellite view. move tail url and options in utils frappe.provide

[Screencast from 23-05-25 06:08:52 PM IST.webm](https://github.com/user-attachments/assets/0d7ea2aa-33b5-4662-b7fa-2f3dbf6f4902)

used open source tails for sattellite view , labels and terrains...

0MUHAMMEDIRFAN committed:
feat(geolocation): add satellite view to input field

Added sattellite view on geolocation input field.
Added labels option and tarrain lines option in sattellite view.
move tail url and options in utils frappe.provide

akhilnarang reviewed (dismissed):
Please setup and run [pre-commit](https://pre-commit.com)

And follow [this](https://github.com/frappe/erpnext/wiki/Coding-Standards#commit-messages) regarding commit messages.

iamejaaz commented on frappe/public/js/frappe/utils/utils.js:
```diff
@@ -1188,10 +1188,31 @@ Object.assign(frappe.utils, {
+					attribution: '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors',
+				}
+			},
+			sattelliteTail: {
```
There's a typo, and this should be in snake_case to stay consistent with the others.
```suggestion
			satellite_tile: {
```

iamejaaz replied:
@0MUHAMMEDIRFAN Changing this alone isn’t enough — you’ll need to update it in other places as well. Also, the naming should follow the same case throughout.

0MUHAMMEDIRFAN replied:
Yes i updated the name "sattelliteTail" to "sattellite_tail" in the whole places. but do i need to use this case in every namings? i have seen that camel case is used in the project and namings , that is why i used the camel case in the namings? shall i need to follow and change the other namings to snake_case?

0MUHAMMEDIRFAN replied:
Changed all other tiles naming to snake_case and fixed the typo

iamejaaz commented on frappe/public/js/frappe/form/controls/geolocation.js:
```diff
@@ -123,7 +124,7 @@ frappe.ui.form.ControlGeolocation = class ControlGeolocation extends frappe.ui.f
 	 * @param {Object} layer - The leaflet layer object.
 	 */
-	on_each_feature(feature, layer) {}
+	on_each_feature(feature, layer) { }
```
The extra space here is not needed.

0MUHAMMEDIRFAN replied:
ran pre-commit and removed unwanted spaces and minor syntaxes

0MUHAMMEDIRFAN commented:
I just remove the merging develop branch to this branch.
Edited the commit title and message for the "semantic commit" workflow.

0MUHAMMEDIRFAN committed:
feat(geolocation): add satellite to geolocation

Applied new review suggestions
Followed snake_case for consistency
Ran pre-commit

0MUHAMMEDIRFAN committed:
Merge branch 'frappe:develop' into geolocation-modify
